spec/unit/middleware/rate_limiter_spec.rb

```diff
@@ -371,8 +372,7 @@ module Middleware
       end
     end
 
-    RSpec.describe RequestCounter do
-      let(:request_counter) { RequestCounter.instance }
+    RSpec.describe REQUEST_COUNTER do
```

Comment:
This is a test for the `BaseRequestCounter`. So you could define a subject for the tests (`let(:request_counter) { BaseRequestCounter.new }`) and use it for invocations as well as expectations.